components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -1469,6 +1470,11 @@ def getRetvalDeclarationForType(returnType, descriptorProvider):
         if returnType.nullable():
             result = CGWrapper(result, pre="Option<", post=">")
         return result
+    if returnType.isTypedArray() and returnType.tag() in builtinNames and descriptorProvider.interface.identifier.name == 'AudioBuffer':
+        result = CGGeneric(builtinNames[returnType.tag()])
+        if returnType.nullable():
+            result = CGWrapper(result, pre="Option<", post=">")
+        return result
```

Comment:
A couple comments here:
 1. Is the body of this `if` identical to the one above? Can it be merged?
 2. What's the explanation for the special-case for `AudioBuffer`? If this is necessary could it be moved to a separate array of names that need this with some documentation on the array?

Reply:
1. Which one do you mean? 
2. Yes that's for now so that this only affects audio buffer and I can get a first try to compile quickly. I'm thinking to add a list of affected WebIDL interfaces, so that the work can be broken into various PR's and issues(which could be marked as easy), where each one would tick off an interface.

Reply:
It looks like the code starting at line 1469 is identical to the code starting at line 1474?

Reply:
The one above is [isPrimitive](https://github.com/servo/servo/blob/28adcf9f6b43e8051f80356ee8583b09359bf62f/third_party/WebIDL/WebIDL.py#L3795), which covers the ones listed up to `double` at https://github.com/servo/servo/blob/28adcf9f6b43e8051f80356ee8583b09359bf62f/third_party/WebIDL/WebIDL.py#L3588

Reply:
The body of the `isPrimitive()` check is:

```
        result = CGGeneric(builtinNames[returnType.tag()])
        if returnType.nullable():
            result = CGWrapper(result, pre="Option<", post=">")
        return result
```

The body of the new if condition you've added is:

```
        result = CGGeneric(builtinNames[returnType.tag()])
        if returnType.nullable():
            result = CGWrapper(result, pre="Option<", post=">")
        return result
```

These two are the same. It seems like this duplication could be avoided.

Reply:
Ok I see, thanks for pointing it out.